documentation/src/main/asciidoc/topics/ref_query_vector_search.adoc

```diff
@@ -0,0 +1,115 @@
+[id='vector-search_{context}']
+= Vector search queries
+
+You can perform vector kNN searches with the Ickle query language using the special operator `+<->+` to define predicates.
+
+This is an example of kNN query:
+
+[source,sql]
+----
+from play.Item i where i.myVector <-> [7,7,7]~3
+----
+
+This query will find the items that have the `myVector` fields that are within `3` nearest neighbourhood from the vector `[7,7,7]`.
+
+Notice that in order to use this kind of search the entity, in our example `play.Item`, has to be `@Indexed` and
+the field, in our example `myVector`, should be annotated with `@Vector`.
+
+We support two kinds of vector field types:
+
+* `byte` / `Byte` (to work with byte vectors)
+* `float` / `Float` (to work with float vectors)
+
+You can have different vector fields on the same entity, but in any case you can have only one vector predicate on your queries.
+
+== Vector search parameters
+
+Both the k-value and the vector can be passed as query parameter.
+The k-value scalar can be expressed with the usual placeholder `:k` in the Ickle text.
+For the vector we can use either a placeholder for each term of the vector:
+
+[source,java]
+----
+Query<Item> query = cache.query("from play.Item i where i.floatVector <-> [:a,:b,:c]~:d");
+query.setParameter("a", 1);
+query.setParameter("b", 4.3);
+query.setParameter("c", 3.3);
+query.setParameter("d", 4);
+----
+
+Or a placeholder can be used for the entire vector:
+
+[source,java]
+----
+Query<Item> query = cache.query("from play.Item i where i.floatVector <-> [:a]~:b");
+query.setParameter("a", new float[]{7.1f, 7.0f, 3.1f});
+query.setParameter("b", 3);
+----
+
+== Vector field attributes
+
+It is always required to specify the **dimension** of the vector field.
+
+The other mapping attributes are optional, since Infinispan will have a default for each of them.
+You can configure them for instance to tune the desired accuracy / performances.
+
+=== Similarity
+
+Different `VectorSimilarity` algorithms are supported
+
+[%autowidth]
+|===
+| Value | Distance | Score | Note
+
+| `L2`
+|
+ifdef::backend-html5[stem:[d(x,y) = \sqrt{\sum_{i=1}^{n} (x_i - y_i)^2 } ]]
```

Comment:
I remember that for Search, we had to add
```
:stem: latexmath
```
so that the formulas showed up correctly, but maybe here, the setup is different 🙈 
(https://github.com/hibernate/hibernate-search/commit/73caaf29c19b6c60faf4f933f9318c38d0aff579#diff-9193834610bf63b77c002612c7c35b01cc46eb0d195ddc57e4b520bb37949988R11)

Reply:
Compiling the code and see the result it seems that is good also without setting the stem.
Anyway I'm adding since it seems that adding it does not create any trouble.
Actually it seems that the formulas are nicer. Thanks :)